### ISSUE TITLE
[composite] `graph.names()` should show only constituents the logged in user has access to

### ISSUE THREAD

**Issue by ikwattro** [neo4j/neo4j#13215]
The following function : 

```
RETURN graph.names()
```

will return all the constituents of a composite db, even those for which the current logged in user is not granted to.

This exposes two issues : 

1. Security issue showing a user names of databases he doesn't have access to
2. Usability issues for the following query : 

```cypher
UNWIND graph.names() AS graph
CALL { 
  USE graph.byName(graph)
  MATCH (n) RETURN count(n) AS c
}
RETURN c
```

Which would fail with an access denied exception for one of the constituent, making the whole query fail

**Neo4j Version:** 5.8.0-enterprise
**Operating System:** Irrelevant
**API:** Irrelevant

### Expected behavior

`graph.names()` function should return only constituents for which the user has access to

### Actual behavior

It returns all constituents independently from access control rules

**Comment by nadja-muller** [comment]
Thank you for reporting this. We will have a look.

**Comment by LinneaAndersson** [comment]
@ikwattro Thanks for your report. We have been able to reproduce and fix the issue. 

The fix will be part of Neo4j 5.11. Please let us know if you have any other issues. 

//Linnéa
